tests/integration/test_logcollector/test_command_monitoring/test_command_execution_freq.py

```diff
@@ -0,0 +1,115 @@
+    {'log_format': 'full_command', 'command': 'echo full_command frequency 86400', 'frequency': 86400}  # 24 hours.
+]
+
+configurations = load_wazuh_configurations(configurations_path, __name__, params=parameters, metadata=metadata)
+configuration_ids = [f"{x['LOG_FORMAT'], x['COMMAND'], x['FREQUENCY']}" for x in parameters]
+
+
+# fixtures
+@pytest.fixture(scope="module", params=configurations, ids=configuration_ids)
+def get_configuration(request):
+    """Get configurations from the module."""
+    return request.param
+
+
+@pytest.fixture(scope="module")
+def get_local_internal_options():
+    """Get configurations from the module."""
+    return local_internal_options
+
+
+def test_command_execution_freq(get_local_internal_options, configure_local_internal_options, get_configuration,
+                                configure_environment, restart_logcollector):
+    """Check if the Wazuh run correctly with the specified command monitoring option "frequency".
+
+    For this purpose, it is verified that the command has not been executed
+    before the period established in this option.
+
+    Args:
+        get_local_internal_options (fixture): Get internal configuration.
+        configure_local_internal_options (fixture): Set internal configuration.
+        get_configuration (fixture): Get configurations from the module.
+        configure_environment (fixture): Configure a custom environment for testing.
+        restart_logcollector (fixture): Reset log file and start a new monitor.
+
+    Raises:
+        TimeoutError: If the command monitoring callback is not generated.
+        AssertError: In the case of a server instance, the API response is different that the real configuration.
+    """
+    cfg = get_configuration['metadata']
+    log_format_message = 'full command' if cfg['log_format'] == 'full_command' else 'command'
+    msg = fr"DEBUG: Running {log_format_message} '{cfg['command']}'"
+    log_callback = monitoring.make_callback(pattern=msg, prefix=prefix)
+    seconds_to_travel = cfg['frequency'] / 2  # Middle of the command execution cycle.
+
+    wazuh_log_monitor.start(timeout=5, callback=log_callback,
+                            error_message=logcollector.GENERIC_CALLBACK_ERROR_COMMAND_MONITORING)
+
+    before = str(datetime.now())
+    TimeMachine.travel_to_future(timedelta(seconds=seconds_to_travel))
+    logger.debug(f"Changing the system clock from {before} to {str(datetime.now())}")
+
+    # The command should not be executed in the middle of the command execution cycle.
+    with pytest.raises(TimeoutError):
+        wazuh_log_monitor.start(timeout=5, callback=log_callback,
+                                error_message=logcollector.GENERIC_CALLBACK_ERROR_COMMAND_MONITORING)
+
+    before = str(datetime.now())
+    TimeMachine.travel_to_future(timedelta(seconds=seconds_to_travel-5))
+    logger.debug(f"Changing the system clock from {before} to {str(datetime.now())}")
```

Comment:
```suggestion
    logger.debug(f"Changing the system clock from {before} to {datetime.now()}")
```

Reply:
Done in https://github.com/wazuh/wazuh-qa/pull/1257/commits/1e33fb8c458aeb8a738e22706713ec98fbf25703